packages/nuxt/src/pages/runtime/router.ts

```diff
@@ -55,15 +55,16 @@ export default defineNuxtPlugin(async (nuxtApp) => {
   nuxtApp.vueApp.component('NuxtChild', NuxtPage)
 
   const baseURL = useRuntimeConfig().app.baseURL
-  const routerHistory = process.client
-    ? createWebHistory(baseURL)
-    : createMemoryHistory(baseURL)
+  const routerHistory = routerOptions.history?.(baseURL) ??
+    (process.client ? createWebHistory(baseURL) : createMemoryHistory(baseURL))
+
+  const routes = !routerOptions.routes ? _routes : Array.isArray(routerOptions.routes) ? routerOptions.routes : routerOptions.routes(_routes)
```

Comment:
I vote to accept function for now. We might alternatively use a _merging strategy_ when `routes` as the array is provided...

```suggestion
  const routes = !routerOptions.routes ? _routes : routerOptions.routes(_routes)
```

Reply:
oh, I see - you mean support function only 👍